### PR TITLE
feat(ST-82): add classroom form modal with validation using zod

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Created ClassroomFormModal.tsx file.
- Created Zod schema for form validation and stored it in ClassroomFormModal.helpers.ts file.
- Created ClassroomFormModal.constants.ts and ClassroomFormModal.styles.ts file.
- Added the ClassroomFormModal components in the CreateClassroomButton and Navbar file.

## Associated Ticket Link(s)

- https://trello.com/c/uD5QwMR0

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/5b2492c5-cd6c-4b74-a621-b66cabc63949

### CODE REVIEW

File: modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  ClassroomFormData,
+  ClassroomFormDefault,
+  ClassroomFormSchemaResolver,
+} from "./ClassroomFormModal.helpers";
+import { useClassroomFormModalStyles } from "./ClassroomFormModal.styles";
+
+interface ClassroomFormModalProps {
+  opened: boolean;
+  onClose: () => void;
+}
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx:18]: This should be in the types file.

File: modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    formState: { errors },
+    control,
+    reset,
+  } = useForm<ClassroomFormData>({
+    resolver: ClassroomFormSchemaResolver,
+    defaultValues: ClassroomFormDefault,
+  });
+
+  const onSubmit = (data: ClassroomFormData) => {
+    console.log(data);
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx:35]: If this is a placeholder, add a `// TODO` comment

File: modules/Dasboard/components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/Navbar/Navbar.tsx:1]: Do you need the React package imported?

File: modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  const styles = useClassroomFormModalStyles(theme);
+
+  const {
+    handleSubmit,
+    formState: { errors },
+    control,
+    reset,
+  } = useForm<ClassroomFormData>({
+    resolver: ClassroomFormSchemaResolver,
+    defaultValues: ClassroomFormDefault,
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomFormModal/ClassroomFomModal.tsx:31]:
```suggestion
    defaultValues: ClassroomFormDefaultValues,
```

File: modules/Dasboard/components/ClassroomFormModal/ClassroomFormModal.helpers.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+export const ClassroomFormSchema = z.object({
+  title: z.string().min(1, { message: "Title is required" }),
+  subject: z.string().min(1, { message: "Subject is required" }),
+  class_time: z.string().min(1, { message: "Class time is required" }),
+  days: z.array(z.string()).min(1, { message: "At least one day is required" }).optional(),
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/ClassroomFormModal/ClassroomFormModal.helpers.ts:8]: Should this be an optional field?

File: modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx [hunk 6]
```diff
@@ -15,9 +20,11 @@ const CreateClassroomButton: React.FC = () => {
         leftIcon={<FaPlus size={14} />}
         className={classes.button}
         classNames={{ leftIcon: classes.icon }}
+        onClick={handleOpenModal}
       >
         Create a classroom
       </Button>
+      <ClassroomFomModal opened={isModalOpen} onClose={handleCloseModal} />
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx:27]: [SUPER NIT]

```suggestion
      <ClassroomFormModal opened={isModalOpen} onClose={handleCloseModal} />
```

File: modules/Dasboard/components/Navbar/Navbar.tsx (hunk 3 above)
Comment (akib-raihan-sazim) [reply modules/Dasboard/components/Navbar/Navbar.tsx:1]: Bhaiya, this came with the "rafce" command.

File: modules/Dasboard/components/ClassroomFormModal/ClassroomFormModal.helpers.ts (hunk 5 above)
Comment (akib-raihan-sazim) [reply modules/Dasboard/components/ClassroomFormModal/ClassroomFormModal.helpers.ts:8]: No bhaiya, there were some error so I used that but later I forgot to remove it.

File: modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx (hunk 6 above)
Comment (akib-raihan-sazim) [reply modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx:27]: Why was it working ? Even though the spelling was mistake

Comment (ishmam-iqbal-sazim) [reply modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx]: The component name itself was likely wrong and then auto imports did its thing.

File: modules/Dasboard/components/Navbar/Navbar.tsx (hunk 3 above)
Comment (ishmam-iqbal-sazim) [reply modules/Dasboard/components/Navbar/Navbar.tsx:1]: Removing it should be fine.

File: modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx [hunk 7]
```diff
@@ -1,12 +1,17 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (ishmam-iqbal-sazim) [inline modules/Dasboard/components/CreateClassroomButton/CreateClassroomButton.tsx:1]: Can remove the React import

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**